### PR TITLE
Vet Clinic Milestone #5: Database performance audit

### PR DESCRIPTION
**Pull request by karlavdelgadof**
## Vet Clinic: add Database performance audit :watch: :chart_with_upwards_trend:  :hospital: :memo: :arrows_counterclockwise:  

These are the proposed changes :recycle:  for this project :arrow_down: :

- Added an extra column `email` to the owners table :card_file_box: 
- Inserted new [provided data](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_performance_audit.md#:~:text=Run%20the%20following%20statements%20to%20add%20data%20to%20your%20database%20(executing%20them%20might%20take%20a%20few%20minutes)%3A) :1234: 
- Ran provided queries by using `EXPLAIN ANALYZE` to perform an audit :interrobang: :memo: 
- Created database indexes on columns `animal_id`, `vet_id` and `email` to speed up the execution time of the queries :100: 
- Copied the SQL used in the previous points into their respective files in the generated repository. :open_file_folder: 

### Queries' Results: :camera: 

**Performance audit (before)**
performance-audit-bef

**Performance audit (after)**
performance-audit-aft

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -82,4 +82,9 @@ INSERT INTO visits (animal_id, vet_id, visit_date)
       ((SELECT id FROM animals WHERE name = 'Blossom'), (SELECT id FROM vets WHERE name = 'Stephanie Mendez'), '2020-05-24'),
       ((SELECT id FROM animals WHERE name = 'Blossom'), (SELECT id FROM vets WHERE name = 'William Tatcher'), '2021-01-11');
 
+/* Add 3.594.280 visits considering 10 animals, 4 vets, and it will use around 87.000 timestamps */
 
+INSERT INTO visits (animal_id, vet_id, visit_date) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
+
```
Comment (KDlamini) [inline data.sql:88]: I noticed that in one of your **before** screenshots your execution time is significantly less than 1sec (<1000ms)

image

- Kindly try to run the populate database query a few times more as described in the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_performance_audit.md#:~:text=4%20Depening%20on,bigger%20than%201000ms.). Depending on your machine speed, we might need to run this query multiple times to reach at least 1sec (1000ms).

NB: You might need to update your screenshots after successfully completing this task.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @karlavdelgadof, @Jeanbulambo 🙋‍♀️,  

Excellent work! 💯 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (karlavdelgadof) [reply data.sql:88]: @KDlamini Thank you for pointing this out! :+1: Here I inserted more records and ran the performance audit again :heavy_check_mark: 
### Performance audit results (before and after) :camera_flash: 
performance-audit-2

**Review by Munsa1 — APPROVED** [review]
Hi @karlavdelgadof, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (KDlamini) [reply data.sql:88]: @karlavdelgadof Great job 👍🏾👏🏾

**Pull request merged**